### DISCUSSION TITLE
nuxt instance undefined

### DISCUSSION THREAD

**Discussion by marekmis** [vuejs/pinia#1456] [Help and Questions]
I just upgraded my app to latest 
"nuxt": "^3.0.0-rc.6", 
and 
"@pinia/nuxt": "^0.3.0",

When calling this.$nuxt or $nuxt in store actions I get undefined error. 
```
 myAction() {
  console.log(this.$nuxt) // undefined
}
```

 I found this is working:
```
 myAction() {
  const nuxtApp = useNuxtApp()
  console.log(nuxtApp) // Nuxt object
}
```
but I don't like how I have to embed it within every action... Is there a way to define nuxtApp once, globally for the entire store?

**Comment by posva** [comment]
This was one of the breaking changes: https://github.com/vuejs/pinia/blob/v2/packages/nuxt/CHANGELOG.md#030-2022-07-13
You can call `useNuxtApp()` in a [plugin](https://pinia.vuejs.org/core-concepts/plugins.html#plugins) to make it available

**Reply by marekmis** [reply]
Any chance for a code sample of a Pinia plugin exposing $nuxt for interactions? 🙏🥹
